Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1855,8 +1855,8 @@ extension BrowserViewController: TopToolbarDelegate {
     
     // TODO: This logic should be fully abstracted away and share logic from current MenuViewController
     // See: https://github.com/brave/brave-ios/issues/1452
-    func topToolbarDidTapBookmarkButton(_ topToolbar: TopToolbarView) {
-        let vc = BookmarksViewController(folder: nil, isPrivateBrowsing: PrivateBrowsingManager.shared.isPrivateBrowsing)
+    func topToolbarDidTapBookmarkButton(_ topToolbar: TopToolbarView?, favorites: Bool) {
+        let vc = BookmarksViewController(showFavorites: favorites, isPrivateBrowsing: PrivateBrowsingManager.shared.isPrivateBrowsing)
```

Comment:
This view controller needs to be reworked for favorites. For example `new folder` button should be disabled